akka-actor-tests/src/test/scala/akka/actor/ReceiveTimeoutSpec.scala

```diff
@@ -179,15 +179,62 @@ class ReceiveTimeoutSpec extends AkkaSpec {
         context.setReceiveTimeout(500 milliseconds)
 
         def receive = {
-          case TransperentTick => context.setReceiveTimeout(Duration.Inf)
+          case TransparentTick => context.setReceiveTimeout(Duration.Inf)
           case ReceiveTimeout  => timeoutLatch.open
         }
       }))
 
-      timeoutActor ! TransperentTick
+      timeoutActor ! TransparentTick
 
       intercept[TimeoutException] { Await.ready(timeoutLatch, 1 second) }
       system.stop(timeoutActor)
     }
+
+    "remove / change existing timeout while handling a NotInfluenceReceiveTimeout message" taggedAs TimingTest in {
+      val timeoutLatch = TestLatch()
+      val initialTimeout = 500.millis
+
+      val timeoutActor = system.actorOf(Props(new Actor {
+        context.setReceiveTimeout(initialTimeout)
+
+        def receive: Receive = {
+          case TransparentTick => context.setReceiveTimeout(Duration.Undefined)
+          case ReceiveTimeout  => timeoutLatch.open
+        }
+      }))
+
+      timeoutActor ! TransparentTick
+
+      intercept[TimeoutException] { Await.ready(timeoutLatch, initialTimeout * 2) }
+      system.stop(timeoutActor)
+    }
+
+    "work correctly if the timeout is changed multiple times while handling a NotInfluenceReceiveTimeout message" taggedAs TimingTest in {
+      val probe = TestProbe()
+      val initialTimeout = 500.millis
+
+      val timeoutActor = system.actorOf(Props(new Actor {
+        var count = 0
+        context.setReceiveTimeout(initialTimeout)
+
+        def receive: Receive = {
+          case TransparentTick =>
+            context.setReceiveTimeout(initialTimeout * 2)
+            count += 1 // do some work then
+            context.setReceiveTimeout(initialTimeout)
+
+          case ReceiveTimeout => probe.ref ! ReceiveTimeout
+        }
+      }))
+
+      timeoutActor ! TransparentTick
+      timeoutActor ! TransparentTick
+
+      // not triggered by initialTimeout because it was changed by the ticks
+      probe.expectNoMessage(initialTimeout) //  the last set value
+      // now should happen
+      probe.expectMsgType[ReceiveTimeout]
+      system.stop(timeoutActor)
+    }
```

Comment:
What Johannes was looking for was a test that failed before we fixed the bug but is now passing when the bug is fixed. This version passes on master as well.

It can also easily fail here by changing to:
```
probe.expectNoMessage(initialTimeout + 50.millis)
```

which is too tight for CI env and it will have spurious failures.

Here is a version that fails on master and passes here and verifies that the new timeout is set via the TransparentTick.

```scala
    "work correctly if the timeout is changed multiple times while handling a NotInfluenceReceiveTimeout message" taggedAs TimingTest in {
      val probe = TestProbe()
      val initialTimeout = 500.millis

      val timeoutActor = system.actorOf(Props(new Actor {
        var count = 0
        context.setReceiveTimeout(initialTimeout)

        def receive: Receive = {
          case TransparentTick =>
            context.setReceiveTimeout(initialTimeout * 2)
            count += 1 // do some work then
            context.setReceiveTimeout(initialTimeout)

          case ReceiveTimeout => probe.ref ! ReceiveTimeout
        }
      }))

      probe.expectNoMessage(initialTimeout / 2)
      
      timeoutActor ! TransparentTick
      timeoutActor ! TransparentTick

      // not triggered by initialTimeout because it was changed by the ticks
      // wait shorter than last set value, but longer than initialTimeout
      probe.expectNoMessage(initialTimeout / 2 + 50.millis)
      // now should happen
      probe.expectMsgType[ReceiveTimeout]
      system.stop(timeoutActor)
    }
```

Reply:
Thanks @patriknw that makes sense now. Pushed.